app/src/main/java/com/vci/vectorcamapp/intake/presentation/IntakeViewModel.kt

```diff
@@ -122,10 +122,8 @@ class IntakeViewModel @Inject constructor(
                         }
 
                     val monthsSinceIrsResult =
-                        if (surveillanceForm?.wasIrsConducted == true && surveillanceForm.monthsSinceIrs != null) {
-                            intakeValidationUseCases.validateMonthsSinceIrs(surveillanceForm.monthsSinceIrs)
-                        } else {
-                            null
+                        surveillanceForm?.takeIf { it.wasIrsConducted && it.monthsSinceIrs != null }?.let {
+                            intakeValidationUseCases.validateMonthsSinceIrs(it.monthsSinceIrs!!)
```

Comment:
[nitpick]: Using "!!" always scares me, even thought the takeIf guards against this. I would recommend changing this to:

val monthsSinceIrsResult =
                        surveillanceForm?.takeIf { it.wasIrsConducted && it.monthsSinceIrs != null }?.let {
                            it.monthsSinceIrs?.let { months -> intakeValidationUseCases.validateMonthsSinceIrs(months) }
                        }